[ASCII-849] Move few components to the new file hierarchy.

ogaca-dd opened the pull request:
<!--
* New contributors are highly encouraged to read our
  [CONTRIBUTING](/CONTRIBUTING.md) documentation.
* The pull request:
  * Should only fix one issue or add one feature at a time.
  * Must update the test suite for the relevant functionality.
  * Should pass all status checks before being reviewed or merged.
* Commit titles should be prefixed with general area of pull request's change.

-->
### What does this PR do?

This PR move few components to the new [files hierarchy](https://github.com/DataDog/datadog-agent/pull/20140).
It also:
- Rename package `serverDebug` to `serverdebug` (Revive)
- Rename some types `something` to `somethingImpl` to avoid conflict with package name

<!--
* A brief description of the change being made with this pull request.
* If the description here cannot be expressed in a succinct form, consider
  opening multiple pull requests instead of a single one.
-->

### Motivation
The package organisation can be improved because the component definition for example [config.Component](https://github.com/DataDog/datadog-agent/blob/7.47.0/comp/core/config/component.go#L30) is located in the same package as [the implementation of the component](https://github.com/DataDog/datadog-agent/blob/7.47.0/comp/core/config/config.go#L39).
Using config.Component will bring all the packages required for its implementation. Having separate packages would help reducing the coupling between packages and avoid cycle in package imports.
<!--
* What inspired you to submit this pull request?
* Link any related GitHub issues or PRs here.
-->

### Additional Notes

<!--
* Anything else we should know when reviewing?
* Include benchmarking information here whenever possible.
* Include info about alternatives that were considered and why the proposed
  version was chosen.
-->

### Possible Drawbacks / Trade-offs

<!--
* What are the possible side-effects or negative impacts of the code change?
-->

### Describe how to test/QA your changes

<!--
* Write here in detail or link to detailed instructions on how this change can
  be tested/QAd/validated, including any environment setup.
-->

### Reviewer's Checklist
<!--
* Authors can use this list as a reference to ensure that there are no problems
  during the review but the signing off is to be done by the reviewer(s).

Note: Adding GitHub labels is only possible for contributors with write access.
-->

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

ogaca-dd committed:
Move few components to the new file hierarchy.
- Rename package serverDebug to serverdebug (Revive)
- Rename sime types something to somethingImpl to avoid conflict with package name

clamoriniere reviewed (approved):
/lgtm for files owned by @DataDog/container-integrations

ogaca-dd committed:
Merge branch 'main' of github.com:DataDog/datadog-agent into olivierg/update-comp-folder-part2

ogaca-dd commented on comp/systray/systray/component.go:
```diff
@@ -24,8 +19,3 @@ type Params struct {
 }
-
-// Module for ddtray
-var Module = fxutil.Component(
-	fx.Provide(newSystray),
-)
```
The goal of this PR is to have a separate packages between Component and Module (See motivation from the PR description).

hush-hush replied:
Keeping the module here would be a good idea I think to simplify imports. But the `impl` should not be considered "private" as it can offers params, helpers, constructor for groups, ...

ogaca-dd commented on comp/systray/systray/impl/doconfigure.go:
```diff
@@ -4,7 +4,7 @@
 //go:build windows
 
-package systray
+package impl
```
There was some discussions about how to name packages but I am open for improvement.

ogaca-dd committed:
Fix windows build

ogaca-dd committed:
rename pacakge  impl  to componentimpl

ogaca-dd committed:
Add missing file

ogaca-dd committed:
Fix code owner

ogaca-dd committed:
Fix unit tests:
- Revert case for serverdebug (serverdebug -> serverDebug) as this lead to issue on the CI
- Fix Windows build
- Fix Linux build
- Fix revive issues
    - "should use bytes.Equal" gosimple
    - Use NewIconFromHICONForDPI instead of NewIconFromHICON

ogaca-dd committed:
Fix component linter issue
